### PR TITLE
Initialize the Headless CMS Apollo client

### PR DESCRIPTION
**Pull request by nikomakela**
KK-887

Generate the schema code: `yarn generate:graphql`

Use the client like this:
```
  import {
    PageIdType,
    PageQuery,
    usePageQuery,
  } from '../headless-cms/graphql/__generated__';
  import headlessCmsClient from '../headless-cms/client';

  const { data, loading, error } = usePageQuery({
    client: headlessCmsClient,
    variables: {
      id: 'cG9zdDox',
      idType: PageIdType.Id,
    },
  });
```

### CODE REVIEW

**Comment by nikomakela** [comment]
https://github.com/City-of-Helsinki/kukkuu-ui/runs/5633498416?check_suite_focus=true#step:5:158
```
#15 [staticbuilder 2/2] RUN yarn build
#15 2.945 Creating an optimized production build...
#15 87.59 Failed to compile.
#15 87.59 
#15 87.60 /app/src/utils/cmsMockDataUtils.ts
#15 87.60 TypeScript error in /app/src/utils/cmsMockDataUtils.ts(1,19):
#15 87.60 Cannot find module '@faker-js/faker' or its corresponding type declarations.  TS2307
```

I don't understand why the faker package is not found. The faker-team has quit and the new readme is here: https://github.com/faker-js/faker#typescript-support. Can't see what is missing.

**Comment by Jokinen** [comment]
In the build phase, only production dependencies have been installed in the container.

Development dependencies are not available.

If code in the production bundle depends on faker, add it to regular dependencies.

If the file that requires it is purely for testing, try to exclude it from the production bundle--if that's too cumbersome you can add it to regular dependencies.

File: src/domain/api/__tests__/client.test.ts [hunk 1]
```diff
@@ -1,45 +1,45 @@
...
-          },
-          children: { __typename: 'ChildNodeConnection' },
-        },
-      })
+    const fetchMock = jest.spyOn(global, 'fetch').mockImplementation(
+      () =>
+        Promise.resolve({
+          json: () => Promise.resolve(jsonData),
+          text: () => Promise.resolve(JSON.stringify(jsonData)),
+        }) as Promise<Response>
```
Comment (Jokinen) [inline src/domain/api/__tests__/client.test.ts:33]: NIcely done 👍

**Review by Jokinen — APPROVED** [review]

**Review by Jokinen — APPROVED** [review]
Oh, and I would have expected to see `headlessCms` instead of `headless-cms` for the module name.

File: package.json [hunk 2]
```diff
@@ -136,5 +137,13 @@
       "!src/serviceWorker.ts",
       "!src/setupTests.ts"
     ]
+  },
+  "lint-staged": {
```
[the remark this answers is not in the document]
Comment (nikomakela) [reply package.json:140]: accidentally added / copy-paste - I will delete this. I had some hard time with the linter fixer: it does not automatically cut too long comments in generated GraphQL schema file. Any ideas?

File: package.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
   "scripts": {
     "build": "react-scripts build",
     "graphql-types": "apollo client:codegen --target=typescript --no-addTypename --outputFlat src/domain/api/generatedTypes",
+    "generate:graphql": "DOTENV_CONFIG_PATH=\"./.env.development\" graphql-codegen -r dotenv/config --config codegen.yml",
```
[the remark this answers is not in the document]
Comment (nikomakela) [reply package.json:19]: They could be ran together, if they don't need any additional configurations. I'll look that.

This needs fix:
- [ ] `yarn graphql-types` fails at the moment.

**Review by Jokinen — APPROVED** [review]

**Pull request merged**